### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by 17tanya**
JavaScript flintbits: SSH and WinRM

### CODE REVIEW

File: winrm/basic/workflow/ipconfig.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+result = response.get("result");                //Response result
+//regexp = /IPv4 Address.*:/;
+//ind = result.indexOf(regexp);
+//temp_str = result.substring(ind);
+/*-9]{1,3}.[0-9]{1,3}/;
+res_arr = result.match(regexp);
+ipadrs = res_arr[0];
+*/
+
+ind1 = result.indexOf("IPv4 Address");
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/ipconfig.js]: Make use of proper variable names

File: winrm/basic/workflow/systeminfo.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+//WinRM Connector Response's meta parameters
+response_exitcode = response.exitcode();        //Exit status code
+response_message = response.message();          //Execution status message
+
+
+user_message="";
+
+//WinRM Connector Response's Result parameter
+result = response.get("result");                //Response result
+ind_host = result.indexOf("Host Name");         //Extracting information
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/systeminfo.js]: Make use of proper variable names

File: winrm/basic/workflow/systeminfo.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: Find System information on target
+ * Description: Find System information on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/systeminfo.js]: Add exception handling.

File: winrm/basic/workflow/pwd.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: Find Current directory on target
+ * Description: Find current directory on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/pwd.js]: Add exception handling.

File: winrm/basic/workflow/powershell_command.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: To execute a powershell command on cmd on target
+ * Description: To execute a powershell command on cmd  on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/powershell_command.js]: Add exception handling.

File: winrm/basic/workflow/ipconfig.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: Find configuration of network interface on target
+ * Description: Find configuration of network interface on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/ipconfig.js]: Add exception handling.

File: winrm/basic/workflow/hostname.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: Find hostname on target
+ * Description: Find hostname on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/hostname.js]: Add exception handling.

File: winrm/basic/workflow/execute_command.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: Execute any command on target
+ * Description: Execute any command on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/execute_command.js]: Add exception handling.

File: winrm/basic/workflow/dir.js [hunk 9]
```diff
@@ -0,0 +1,128 @@
+/**
+ * Creation Date: 15/05/2019
+ * Summary: To list folders and files in the present working directory
+ * Description: list folders and files in the present working directory on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/dir.js]: Add exception handling.

File: winrm/basic/workflow/date.js [hunk 10]
```diff
@@ -0,0 +1,137 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: Find date on target
+ * Description: Find date on target using WinRM Protocol
+*/
+
```
Comment (atulgirishkumar) [inline winrm/basic/workflow/date.js]: Add exception handling.

File: ssh/operation/workflow/systeminfo2.js [hunk 11]
```diff
@@ -0,0 +1,226 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: to view present working directory on target
+ * Description: To view present working directory on target using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/systeminfo2.js]: Add exception handling.

File: ssh/operation/workflow/systeminfo.js [hunk 12]
```diff
@@ -0,0 +1,170 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: to view present working directory on target
+ * Description: To view present working directory on target using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/systeminfo.js]: Add exception handling.

File: ssh/operation/workflow/ssh.js [hunk 13]
```diff
@@ -0,0 +1,164 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: Execute any command on target
+ * Description: Execute any command on target using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/ssh.js]: Add exception handling.

File: ssh/operation/workflow/pwd.js [hunk 14]
```diff
@@ -0,0 +1,167 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: to view present working directory on target
+ * Description: To view present working directory on target using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/pwd.js]: Add exception handling.

File: ssh/operation/workflow/processor_info.js [hunk 15]
```diff
@@ -0,0 +1,192 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: System information on target
+ * Description: To view System information on target using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/processor_info.js]: Add exception handling.

File: ssh/operation/workflow/ifconfig.js [hunk 16]
```diff
@@ -0,0 +1,179 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: View configuration of network interface on target
+ * Description: To view config of the network interface on target using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/ifconfig.js]: Add exception handling.

File: ssh/operation/workflow/hostname.js [hunk 17]
```diff
@@ -0,0 +1,168 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: Find Hostname on the target
+ * Description: To find Hostname on the target machine using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/hostname.js]: Add exception handling.

File: ssh/operation/workflow/date.js [hunk 18]
```diff
@@ -0,0 +1,170 @@
+/**
+ * Creation Date: 14/05/2019
+ * Summary: Find Date on the target
+ * Description: To find Date on the target machine using SSH Protocol
+*/
+
```
Comment (atulgirishkumar) [inline ssh/operation/workflow/date.js]: Add exception handling.

File: ssh/operation/workflow/date.js [hunk 19]
```diff
@@ -0,0 +1,170 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                               //Not mandatory as authentication type
+    log.trace("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/date.js]: Made the necessary changes.

File: ssh/operation/workflow/hostname.js [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                               //Not mandatory as authentication type 
+    log.error("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/hostname.js]: Resolved.

File: ssh/operation/workflow/ifconfig.js [hunk 21]
```diff
@@ -0,0 +1,179 @@
...
+else{                                           //Not mandatory as authentication type
+    log.trace("Passphrase not given");          //depends on user
+}
+
+
+//connector call
+
+
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/ifconfig.js]: Resolved.

File: ssh/operation/workflow/processor_info.js [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                                //Not mandatory: authentication type
+    log.trace("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/processor_info.js]: Resolved.

File: ssh/operation/workflow/pwd.js [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                               //Not mandatory authentication type
+    log.trace("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/pwd.js]: Resolved.

File: ssh/operation/workflow/ssh.js [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                               //Not mandatory authentication type
+    log.trace("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/ssh.js]: Resolved.

File: ssh/operation/workflow/systeminfo.js [hunk 25]
```diff
@@ -0,0 +1,170 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                               //Not mandatory authentication type
+    log.trace("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/systeminfo.js]: Resolved.

File: ssh/operation/workflow/systeminfo2.js [hunk 26]
```diff
@@ -0,0 +1,226 @@
...
+    log.info("passphrase:"+passphrase);
+}
+else{                                               //Not mandatory authentication type
+    log.trace("Passphrase not given");              //depends on user
+}
+
+
+//connector call
+// For password-based authentication
+if(password!=null || password!=""){
```
[the remark this answers is not in the document]
Comment (17tanya) [reply ssh/operation/workflow/systeminfo2.js]: Resolved.

File: winrm/basic/workflow/ipconfig.js (hunk 1 above)
Comment (17tanya) [reply winrm/basic/workflow/ipconfig.js]: Variable names changed.

File: winrm/basic/workflow/systeminfo.js (hunk 2 above)
Comment (17tanya) [reply winrm/basic/workflow/systeminfo.js]: Variable names changed.

File: winrm/basic/workflow/systeminfo.js (hunk 3 above)
Comment (17tanya) [reply winrm/basic/workflow/systeminfo.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/pwd.js (hunk 4 above)
Comment (17tanya) [reply winrm/basic/workflow/pwd.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/powershell_command.js (hunk 5 above)
Comment (17tanya) [reply winrm/basic/workflow/powershell_command.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/ipconfig.js (hunk 6 above)
Comment (17tanya) [reply winrm/basic/workflow/ipconfig.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/hostname.js (hunk 7 above)
Comment (17tanya) [reply winrm/basic/workflow/hostname.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/execute_command.js (hunk 8 above)
Comment (17tanya) [reply winrm/basic/workflow/execute_command.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/dir.js (hunk 9 above)
Comment (17tanya) [reply winrm/basic/workflow/dir.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: winrm/basic/workflow/date.js (hunk 10 above)
Comment (17tanya) [reply winrm/basic/workflow/date.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/systeminfo2.js (hunk 11 above)
Comment (17tanya) [reply ssh/operation/workflow/systeminfo2.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/systeminfo.js (hunk 12 above)
Comment (17tanya) [reply ssh/operation/workflow/systeminfo.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/ssh.js (hunk 13 above)
Comment (17tanya) [reply ssh/operation/workflow/ssh.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/pwd.js (hunk 14 above)
Comment (17tanya) [reply ssh/operation/workflow/pwd.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/processor_info.js (hunk 15 above)
Comment (17tanya) [reply ssh/operation/workflow/processor_info.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/ifconfig.js (hunk 16 above)
Comment (17tanya) [reply ssh/operation/workflow/ifconfig.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/hostname.js (hunk 17 above)
Comment (17tanya) [reply ssh/operation/workflow/hostname.js]: Exception handling is not required as mentioned by Abhishek Sir.

File: ssh/operation/workflow/date.js (hunk 18 above)
Comment (17tanya) [reply ssh/operation/workflow/date.js]: Exception handling is not required as mentioned by Abhishek Sir.

**Pull request merged**